src/main/scala/abalone/variant/Variant.scala

```diff
@@ -98,8 +98,8 @@ abstract class Variant private[variant] (
     situation.board.piecesOf(situation.player).flatMap {
       case ((pos, piece)) =>
         Map(pos ->
-          Stone.dirs.flatMap { _(pos) } // if the piece would remain on the board after applying a direction...
-          .filterNot(situation.board.pieces.contains(_)) // ...keep it only if the landing square is free of marble
+          pos.neighbours.flatten // for each direction that would remain on the board...
+          .filterNot(situation.board.pieces.contains(_)) // ...keep the piece only if the landing square is empty
```

Comment:
do you want me to reword as we keep a pos, not a piece ?
even if it's a case of pos having the piece on top of it ?

Reply:
I don't think the comments are necessary, its clear from the code/function names what is happening. The function return type tells us what we are creating.

Reply:
yes I agree I should avoid comments explaining the scala code itself, even if the lines may have been not so clear instantly for me, someone with more experience will understand very easily such simple line, I removed it thanks